### ISSUE TITLE
Segfault calling `simple-selectors()`

### ISSUE THREAD

**Issue by chriseppstein** [sass/node-sass#1880]
`simple-selectors('&:hover')` causes a segfault.

In ruby Sass it gives an error:

```
Error: $selector: "&:hover" is not a valid selector: Invalid CSS after "": expected selector, was "&:hover" for `simple-selectors'
        on line 1 of standard input
```

**Comment by xzyfer** [comment]
Which version of node-sass?
Which architecture?

**Comment by xzyfer** [comment]
We're having all kinds of weird environment specific segfaults after the recent memory manager changes in LibSass.

**Comment by chriseppstein** [comment]
```
$ node-sass -v
node-sass	4.5.0	(Wrapper)	[JavaScript]
libsass  	3.5.0.beta.2	(Sass Compiler)	[C/C++]
$  uname -a
Darwin ceppstei-mn1 15.6.0 Darwin Kernel Version 15.6.0: Mon Jan  9 23:07:29 PST 2017; root:xnu-3248.60.11.2.1~1/RELEASE_X86_64 x86_64
```

**Comment by mgreter** [comment]
@xzyfer IMO so far we haven't had one real issue with the memory manager changes. This one is simply a null pointer access leading to a segfault. Fixed in https://github.com/sass/libsass/pull/2318.

**Comment by xzyfer** [comment]
@mgreter https://github.com/sass/node-sass/issues/1879 is related to the memory manager changes.

**Comment by xzyfer** [comment]
Fixed in 4.8.0
